### ISSUE TITLE
Dot crosshair disappears on object interaction

### ISSUE THREAD

**Issue by ISuckM8** [tunbridgep/gmdx-vrsd-fork#30]
When you right click frob a datacube, the crosshair seems to be disabled until you equip and unequip a weapon.

Possible causes: no script to take into account activating the crosshair after interactions.

**Comment by tunbridgep** [comment]
The crosshair updates it status based on the `UpdateCrosshair()` function in `DeusExPlayer.u`

InformationDevices (which Data Cubes inherit from) have code in them to update the crosshair when the display window is closed. They also update it when the window is opened.

From `InformationDevices.uc`:

```
function DestroyWindow()
{
    local DeusExPlayer player;

	// restore the crosshairs and the other hud elements
	if (aReader != None)
	{
        aReader.UpdateCrosshair();
		DeusExRootWindow(aReader.rootWindow).hud.frobDisplay.Show();
	}

	if (infoWindow != None)
	{
		infoWindow.ClearTextWindows();
		infoWindow.Hide();
	}

	infoWindow = None;
	winText = None;
	aReader = None;
}
```

The `UpdateCrosshair()` function in `DeusExPlayer` calls the `UpdateCrosshair` function in `DeusExHUD.uc`

```
function UpdateCrosshair(DeusExPlayer player)
{
	cross.SetCrosshair(player.GetCrosshairState());
	hitmarker.SetCrosshair(player.GetHitMarkerState());
}
```

The important bit is that this shows or hides the crosshair based on the `GetCrosshairState()` function, in DeusExPlayer, which checks for a whole bunch of things.

Here's the relevant code from `GetCrosshairState()`:

```
    if (root != None && root.WindowStackCount() > 0) //No crosshair while windows are open
        return false;
```

https://github.com/tunbridgep/gmdx-vrsd-fork/assets/31608726/8c33992e-556a-4944-976a-6c502badc6c9
This video confirms the behaviour is working correctly in the general case.

This means that the issue is likely being caused by either:
1. The window stack count somehow remaining greater than zero when we stop reading the datacube
2. the `UpdateCrosshair()` function isn't being called
3. Something else entirely.

Either way, this seems like a difficult to reproduce issue and will likely be difficult to fix.

**Comment by ISuckM8** [comment]
I'm guessing the datacube closing because of the player looking away is also accounted for? Otherwise, this is very weird indeed.

**Comment by tunbridgep** [comment]
> I'm guessing the datacube closing because of the player looking away is also accounted for? Otherwise, this is very weird indeed.

Yes. In fact the video I posted shows me looking away and not closing the datacube with right click, many times.

`DestroyWindow` is called whether you walk away or close it manually. So the code paths should be the same.

**Comment by tunbridgep** [comment]
I believe this is fixed now.
